charmcraft/extensions/gunicorn.py

```diff
@@ -0,0 +1,238 @@
+                )
+        for merging in ("actions", "requires", "provides", "config.options"):
+            user_provided: dict[str, Any] = self._get_nested(self.yaml_data, merging)
+            if not user_provided:
+                continue
+            overlap = user_provided.keys() & self._get_nested(root_snippet, merging).keys()
+            if overlap:
+                raise ExtensionError(
+                    f"overlapping keys {overlap} in {merging} of charmcraft.yaml "
+                    f"which conflict with the {self.framework}-framework extension, "
+                    "please rename or remove it"
+                )
+        for config in self._get_nested(self.yaml_data, "config.options"):
+            for reserved_config_prefix in ("webserver-", f"{self.framework}-"):
+                if config.startswith(reserved_config_prefix):
+                    raise ExtensionError(
+                        f"config.options {config!r} starts with {self.framework}-framework"
+                        f" reserved configuration prefix {reserved_config_prefix!r}, "
+                        "please rename or remove it"
+                    )
+
+    def _get_root_snippet(self) -> dict[str, Any]:
+        """Return the root snippet to be merged into the user charmcraft.yaml.
+
+        This method differs from get_root_snippet because it doesn't perform any check.
+        """
+        return {
+            "assumes": ["k8s-api"],
+            "containers": {
+                f"{self.framework}-app": {"resource": f"{self.framework}-app-image"},
+            },
+            "resources": {
+                f"{self.framework}-app-image": {
+                    "type": "oci-image",
+                    "description": f"{self.framework} application image.",
+                },
+            },
+            "peers": {"secret-storage": {"interface": "secret-storage"}},
+            "actions": self.actions,
+            "requires": {
+                "logging": {"interface": "loki_push_api"},
+                "ingress": {"interface": "ingress", "limit": 1},
+            },
+            "provides": {
+                "metrics-endpoint": {"interface": "prometheus_scrape"},
+                "grafana-dashboard": {"interface": "grafana_dashboard"},
+            },
+            "config": {"options": {**self._WEBSERVER_OPTIONS, **self.options}},
+            "parts": {"charm": {"plugin": "charm", "source": "."}},
+        }
+
+    @override
+    def get_root_snippet(self) -> dict[str, Any]:
+        """Fill in some required root components."""
+        self._check_input()
+        return self._get_root_snippet()
+
+    @override
+    def get_part_snippet(self) -> dict[str, Any]:
+        """Return the part snippet to apply to existing parts."""
+        return {}
+
+    @override
+    def get_parts_snippet(self) -> dict[str, Any]:
+        """Return the parts to add to parts."""
+        return {}
+
+
+class FlaskFramework(_GunicornBase):
```

Comment:
I think this will need to override the `is_experimental` and set it to `False` since flask already is no longer experimental 🙂

Reply:
Okay, updated.